client/db/src/lib.rs

```diff
@@ -481,10 +489,11 @@ pub struct BlockchainDb<Block: BlockT> {
 	leaves: RwLock<LeafSet<Block::Hash, NumberFor<Block>>>,
 	header_metadata_cache: Arc<HeaderMetadataCache<Block>>,
 	header_cache: Mutex<LinkedHashMap<Block::Hash, Option<Block::Header>>>,
+	delayed_pruning: Option<u32>,
```

Comment:
This should rater be called `delay_canonicalization`

Reply:
Actually, I feel that may be confusing because we also have a [Backend::canonicalization_delay](https://github.com/paritytech/substrate/blob/49734dd1d72a00b9d3b87ba397661a63e0e17af3/client/db/src/lib.rs#L1030). To not be confused with `BlockchainDb::delay_canonicalization` added here.

I initially added the `delayed_pruning` to get an offset for what to report in `BlockchainDb::displaced_leaves_after_finalizing`. 

What do you think about changing it to `displaced_leaves_offset`?